.github/workflows/upstream-dev-ci.yaml

```diff
@@ -11,10 +11,30 @@ on:
   workflow_dispatch: # allows you to trigger the workflow run manually
 
 jobs:
+  detect-ci-trigger:
+    name: detect upstream-dev ci trigger
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    outputs:
+      triggered: ${{ steps.detect-trigger.outputs.trigger-found }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/detect-ci-trigger
+        id: detect-trigger
+        with:
+          keyword: "test-upstream"
```

Comment:
should we require `[test-upstream]`, or is `test-upstream` enough? Not a strong opinion, but `[skip-ci]` seems to be pretty common.

Reply:
> should we require [test-upstream], or is test-upstream enough?

I'm in favor of `[test-upstream]`. I'm going to update this shortly....